pandas/core/arrays/categorical.py

```diff
@@ -2279,7 +2279,7 @@ def _from_factorized(cls, uniques, original):
             original.categories.take(uniques), dtype=original.dtype
         )
 
-    def equals(self, other):
+    def equals(self, other) -> bool:
```

Comment:
```suggestion
    def equals(self, other: Any) -> bool:
```

Purely stylistic but we've been typing `other` as Any elsewhere, so good to do here as well (applicable in a few spots)

Reply:
the docstring says other is Categorical; id prefer not to introduce mismatches